addons/mail/models/mail_thread.py

```diff
@@ -2596,7 +2596,16 @@ def _notify_classify_recipients(self, recipient_data, model_name):
 
         groups = self._notify_get_groups()
 
-        access_link = self._notify_get_action_link('view')
+        link_args = {}
+        if 'msg_vals' in kwargs:
+            msg_vals = kwargs.get('msg_vals')
+            if type(msg_vals) == dict:
+                if 'model' in msg_vals:
+                    link_args['model'] = msg_vals.get('model')
+                if 'res_id' in msg_vals:
+                    link_args['res_id'] = msg_vals.get('res_id')
+
```

Comment:
At this point msg_vals should be either False or a dict, so there is no need to check the type.
```diff
link_args = {}
msg_vals = kwargs.get('msg_vals')
if msg_vals and 'model' in msg_vals and 'res_id' in msg_vals:
    link_args['model'] = msg_vals.get('model')
    link_args['res_id'] = msg_vals.get('res_id')
```